src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +23,32 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{todo?.id}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => onTodoClear()}
+            />
```

Comment:
The close button for the modal does not have an associated label, which can affect accessibility. Consider adding an `aria-label` attribute to this button to provide context for screen reader users.